### PR TITLE
Use world units for point sizes

### PR DESCRIPTION
**Pull request by hhsaez**
By default, <Points /> uses the `scale` property to determine the size of all points. `gl_PointSize` expects values in pixels rather than in units in world space, though.

The changes in this commit introduce a new optional flag to use world space point sizes instead of pixels. In order to accomplish that goal, the vertex shader computes a scaled point size based on the current view transformation and render target size.

Test plan: Added new stories for `<Points />` command, using both screen and world space point sizes with unit cubes for reference.

## Versioning impact

Minor: the new flag is used to ensure backward compatibility

### CODE REVIEW

File: packages/regl-worldview/src/commands/Points.js [hunk 1]
```diff
@@ -6,64 +6,100 @@
...
+        // Then, we compute the distance between both points in clip space, dividing
+        // by the w-component to account for distance in perspective projection.
+        float d = length(up.xyz / up.w - gl_Position.xyz / gl_Position.w);
+
+        // Finally, the point size is calculated using the size of the render target
+        // and it's aspect ratio. We multiply it by 0.5 since distance in clip space
+        // is in range [0, 2] (because clip space's range is [-1, 1]) and
+        // we need it to be [0, 1].
+        float invAspect = viewportHeight / viewportWidth;
+        gl_PointSize = pointSize * 0.5 * d * viewportWidth * invAspect;
```
Comment (troygibb) [inline packages/regl-worldview/src/commands/Points.js:63]: great comments!

File: docs/src/4.8.Points.mdx [hunk 2]
```diff
@@ -6,9 +6,10 @@ import { Points } from './jsx/allLiveEditors'
 
 ## Props
 
-| Name       | Type      | Default | Description                      |
-| ---------- | --------- | ------- | -------------------------------- |
-| `children` | `Point[]` | `[]`    | array of Point markers to render |
+| Name                | Type      | Default        | Description                           |
+| ------------------- | --------- | -------------- | ------------------------------------- |
+| `children`          | `Point[]` | `[]`           | array of Point markers to render      |
+| `useWorldSpaceSize` | `boolean` | `undefined`    | use world space units for point sizes |
```
Comment (troygibb) [inline docs/src/4.8.Points.mdx:12]: might be worth adding a bit how this relates to `scale`

**Review by troygibb — APPROVED** [review]
Nice work here. 

I guess the part that's left is figuring out how to integrate this with the rest of Webviz, any ideas here? I.e. Jason's [issue here](https://github.com/cruise-automation/webviz/issues/491) is rendering points from `MarkerArray`.

**Comment by hhsaez** [comment]
> I guess the part that's left is figuring out how to integrate this with the rest of Webviz, any ideas here? I.e. Jason's [issue here](https://github.com/cruise-automation/webviz/issues/491) is rendering points from `MarkerArray`.

The optional `useWorldSpaceSize ` flag allows Worldview to remain generic, supporting both render modes (screen and world space), since it doesn't really know about markers. In Webviz, we need to use that flag so point markers will be always rendered in world space. This is a similar to how we handle screen invariance in `<GLText />`, which is always enabled in Webviz.

Does that make sense?

**Comment by troygibb** [comment]
> The optional useWorldSpaceSize flag allows Worldview to remain generic, supporting both render modes (screen and world space), since it doesn't really know about markers. In Webviz, we need to use that flag so point markers will be always rendered in world space. This is a similar to how we handle screen invariance in <GLText />, which is always enabled in Webviz.
> Does that make sense?

Yea totally! I guess my real question here is whether adding this flag on by default in Webviz will be backwards compatible with many of the points we already display. Lidar points seem easy, since we control how they're rendered. However, for example, I don't have a good sense how much folks rely on `MarkerArray` points in production.

**Pull request merged**